x/ccv/provider/keeper/punish_validator.go

```diff
@@ -0,0 +1,106 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	evidencetypes "github.com/cosmos/cosmos-sdk/x/evidence/types"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/cosmos/interchain-security/v3/x/ccv/provider/types"
+)
+
+// JailAndTombstoneValidator jails and tombstones the validator with the given provider consensus address
+func (k Keeper) JailAndTombstoneValidator(ctx sdk.Context, providerAddr types.ProviderConsAddress) error {
+	validator, found := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerAddr.ToSdkConsAddr())
+	if !found {
+		return errorsmod.Wrapf(slashingtypes.ErrNoValidatorForAddress, "provider consensus address: %s", providerAddr.String())
+	}
+
+	if validator.IsUnbonded() {
```

Comment:
It would seem to me that this check is not needed here because we already perform the check in `SlashValidator` and we always call `SlashValidator` before calling `JailAndTombstoneValidator`. 

Nothing wrong in living it here but then we could maybe add a comment like:
```
        // we check if the validator is unbonded to prevent panicking when slashing (see cosmos/cosmos-sdk/blob/v0.47.5/x/staking/keeper/slash.go#L61)
        // note that we already check this in `SlashValidator` but we check here as well as a potential future safe guard
        // in case someone calls `JailAndTombstoneValidator` before `SlashValidator`
```